src/rules/valid-expect.ts

```diff
@@ -0,0 +1,297 @@
+          context.report({
+            loc: {
+              end: {
+                column: node.loc.start.column + expectLength + 1,
+                line: node.loc.start.line,
+              },
+              start: {
+                column: node.loc.start.column + expectLength,
+                line: node.loc.start.line,
+              },
+            },
+            messageId: 'noArgs',
+            node,
+          });
+        }
+
+        // something was called on `expect()`
+        if (!matcher) {
+          if (modifier) {
+            context.report({
+              data: { propertyName: modifier.name }, // todo: rename to 'modifierName'
+              messageId: 'propertyWithoutMatcher', // todo: rename to 'modifierWithoutMatcher'
+              node: modifier.node.property,
+            });
+          }
+
+          return;
+        }
+
+        if (matcher.node.parent && isExpectMember(matcher.node.parent)) {
+          context.report({
+            messageId: 'invalidProperty', // todo: rename to 'invalidModifier'
+            data: { propertyName: matcher.name }, // todo: rename to 'matcherName' (or modifierName?)
+            node: matcher.node.property,
+          });
+
+          return;
+        }
+
+        if (!matcher.arguments) {
+          context.report({
+            data: { propertyName: matcher.name }, // todo: rename to 'matcherName'
+            messageId: 'matcherOnPropertyNotCalled', // todo: rename to 'matcherNotCalled'
+            node: matcher.node.property,
+          });
+        }
+
+        const parentNode = matcher.node.parent;
+
+        if (
+          !modifier ||
+          !parentNode ||
+          !parentNode.parent ||
+          modifier.name === ModifierName.not
+        ) {
+          return;
+        }
+        /**
+         * If parent node is an array expression, we'll report the warning,
+         * for the array object, not for each individual assertion.
+         */
+        const isParentArrayExpression =
+          parentNode.parent.type === AST_NODE_TYPES.ArrayExpression;
+        const orReturned = alwaysAwait ? '' : ' or returned';
+        /**
+         * An async assertion can be chained with `then` or `catch` statements.
+         * In that case our target CallExpression node is the one with
+         * the last `then` or `catch` statement.
+         */
+        const targetNode = getParentIfThenified(parentNode);
+        const finalNode =
+          findPromiseCallExpressionNode(targetNode) || targetNode;
+        if (
+          finalNode.parent &&
+          // If node is not awaited or returned
+          !isAcceptableReturnNode(finalNode.parent, !alwaysAwait) &&
+          // if we didn't warn user already
+          !promiseArrayExceptionExists(finalNode.loc)
+        ) {
+          context.report({
+            loc: finalNode.loc,
+            data: {
+              orReturned,
+            },
+            messageId:
+              finalNode !== targetNode
```

Comment:
flip so `true` check is first. I've pushed `no-negated-condition` to master which will make this a lint error

Reply:
but it looks weird having the longer string at the bottom...

Reply:
most stable foundation! 😀